cupy/core/_routines_indexing.pyx

```diff
@@ -62,7 +72,7 @@ cdef tuple _ndarray_nonzero(ndarray self):
         count_nonzero = int(scan_index[-1])  # synchronize!
     ndim = max(<int>self._shape.size(), 1)
     if count_nonzero == 0:
-        return (ndarray((0,), dtype=dtype),) * ndim
+        return ndarray((0,ndim), dtype=dtype)
```

Comment:
CI is failing due to a linter error here. Would suggest adding a space. Should fix it 🙂

```suggestion
        return ndarray((0, ndim), dtype=dtype)
```

Reply:
By failures I meant the pytest failures. The results for numpy and cupy are different.

Reply:
below is traceback to one test failure,

```
self = <cupy.testing._bundle.TestArgwhere_param_0_{array=array([0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1])} testMethod=test_argwhere>
args = (), kw = {'dtype': <class 'numpy.float64'>}
cupy_result = (array([ 2,  3,  4,  5,  7,  8, 10, 12, 13, 18, 19]),)
cupy_error = None, cupy_tb = None
numpy_result = (array([[ 2],
       [ 3],
       [ 4],
       [ 5],
       [ 7],
       [ 8],
       [10],
       [12],
       [13],
       [18],
       [19]]),)
```

Can you take a look please? @jakirkham

Reply:
I even have defined dst as `ndarray(count_nonzero, 1)`. Can't figure out why the resulting dims are (1, count_nonzero). And same is with the array haveing `ndim > 1`. the resulting arrays are transpose of the expected arrays.

Reply:
Yeah I think that is expected. This is what they say about it in the NumPy docs

> `np.argwhere(a)` is the same as `np.transpose(np.nonzero(a))`.

ref: https://docs.scipy.org/doc/numpy/reference/generated/numpy.argwhere.html

Reply:
The kernel in CuPy may be tailored for the `np.nonzero` case. It might be possible to tweak it to align more closely to `np.argwhere`.

Reply:
Maybe [this comment]( https://github.com/cupy/cupy/pull/3135#discussion_r388183512 ) also helps? 🙂

Reply:
(just a note: above failure was of `argwhere` not `nonzero`.)

Reply:
Thanks for the clarification. Maybe we are not seeing that issue given the way the CuPy kernel is constructed.

As to `nonzero`, maybe this is related to 1-D special casing as outlined in [this comment]( https://github.com/cupy/cupy/pull/3135#discussion_r388194084 )?